ow_faults_detection/include/ow_faults_detection/FaultDetector.h

```diff
@@ -85,32 +81,23 @@ class FaultDetector
   ros::Publisher m_power_faults_msg_pub;
   ros::Publisher m_system_faults_msg_pub;
 
-  // Arm and antenna
-  ros::Subscriber m_joint_states_sub;
-
-  // Camera
+  // faults topic subscribers;
+  ros::Subscriber m_joint_states_sub; // for arm and antenna
   ros::Subscriber m_camera_original_trigger_sub;
   ros::Subscriber m_camera_raw_sub;
-
-  // Power
   ros::Subscriber m_power_soc_sub;
   ros::Subscriber m_power_temperature_sub;
 
-  // VARIABLES
-  
-  // System 
-  uint64_t m_system_faults_flags = 0;
+  // faults bitflags
+  uint64_t m_arm_faults_flags     = owl_msgs::ArmFaultsStatus::NONE;
+  uint64_t m_antenna_faults_flags = owl_msgs::PanTiltFaultsStatus::NONE;
+  uint64_t m_camera_faults_flags  = owl_msgs::CameraFaultsStatus::NONE;
+  uint64_t m_power_faults_flags   = owl_msgs::PowerFaultsStatus::NONE;
+  uint64_t m_system_faults_flags  = owl_msgs::SystemFaultsStatus::NONE;
+
+  // component variables
   std::vector<unsigned int> m_joint_state_indices;
-  
-  // Antenna
-  bool m_pan_fault;
-  bool m_tilt_fault;
-  
-  // Camera
   bool m_camera_data_pending = false;
-  
-  // Power
-  uint64_t m_power_faults_flags = 0;
   float m_last_soc = std::numeric_limits<float>::quiet_NaN();
 };
 
```

Comment:
Minor: GitHub won't let me comment on the relevant line because it's not part of the diff.  But `~FaultDetector(){}` could be replaced with `~FaultDetector() = default;` which is bit more proper.